lib/pages/other/WebView.dart

```diff
@@ -7,37 +9,62 @@ import 'package:app/assets/components/customAppBar.dart';
 import 'package:stacked/stacked.dart';
 import 'package:app/viewmodels/web_view_model.dart';
 
-class WebViewArgumnets {
+import '../../locator.dart';
+
+class WebViewArguments {
   final String initialUrl;
 
-  WebViewArgumnets(this.initialUrl);
+  WebViewArguments(this.initialUrl);
 }
 
-class WebViewPage extends StatelessWidget {
-  final WebViewArgumnets args;
+class WebViewPage extends StatefulWidget {
+  final WebViewArguments args;
   WebViewPage(this.args);
+  
+
+  @override
+  _WebViewPageState createState() => _WebViewPageState();
+}
+
+class _WebViewPageState extends State<WebViewPage> {
+
+ final NavigationService _navigationService = locator<NavigationService>();
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: CustomAppBar(
+      appBar: customAppBar(
         backButtonText: "Back",
         context: context,
+        actions: <Widget>[
+         TextButton(
+           child: Icon(Icons.more_vert, color: Theme.of(context).primaryColor),
+           onPressed: ((){
+               _navigationService.launchLink(
+                widget.args.initialUrl,
+                isExternal: true,
+                title: 'Open with an External Browser?',
+                description: 'You are about to open this link with an external browser.',
+                buttonText: 'Yes, Go Ahead',
+                closeButtonText: 'Stay Here'
+                );
+              
+           }),
+           ),
+        ]
       ),
       body: SafeArea(
-        child: ViewModelBuilder<WebViewViewModel>.reactive(
-            viewModelBuilder: () => WebViewViewModel(),
-            builder: (context, model, child) {
-              return WebView(
-                javascriptMode: JavascriptMode.unrestricted,
-                initialUrl: args.initialUrl,
-                onWebResourceError: (error) {
-                  model.onWebError(args.initialUrl);
-                },
-              );
-            }
-          )
-      ),
+          child: ViewModelBuilder<WebViewViewModel>.reactive(
+              viewModelBuilder: () => WebViewViewModel(),
+              builder: (context, model, child) {
+                return WebView(
+                  javascriptMode: JavascriptMode.unrestricted,
+                  initialUrl: widget.args.initialUrl,
+                  onWebResourceError: (error) {
+                    model.onWebError(widget.args.initialUrl);
```

Comment:
This is what I'm talking about in the above comment in terms of using the ViewModel. Once the viewmodel has the url in it we will no longer need to provider the url here either. 
 ```suggestion
                     model.onWebError();
```